src/service/metrics/otlp_grpc.rs

```diff
@@ -218,6 +218,7 @@ pub async fn handle_grpc_request(
                     }
                 }
                 for mut rec in records {
+                    // TODO: What to do here?
```

Comment:
@hengfeiyang any idea how to apply vrl functions with `apply_before_flattening` flag here? The problem in this case is that after flattening we check if the `local_metric_name` and the `metric_name` are equal or not. If not equal we again fetch the vrl functions for the `local_metric_name` and apply it.

But how to know before flattening if `local_metric_name` is same as `metric_name`? The `NAME_LABEL` may not be present before flattening in some cases.

Reply:
Same with `metrics/otlp_http.rs`